src/modules/land_detector/MulticopterLandDetector.h

```diff
@@ -111,10 +111,14 @@ class MulticopterLandDetector : public LandDetector
 	uORB::Subscription _actuator_controls_sub{ORB_ID(actuator_controls_0)};
 	uORB::Subscription _hover_thrust_estimate_sub{ORB_ID(hover_thrust_estimate)};
 	uORB::Subscription _trajectory_setpoint_sub{ORB_ID(trajectory_setpoint)};
+	uORB::Subscription _offboard_trajectory_setpoint_sub {ORB_ID(offboard_trajectory_setpoint)};
```

Comment:
I am not very comfortable from the fact that this is creeping into the land detector, which should not need to be aware whether the vehicle is in offboard or not. Would there be another way to handle this case?

Reply:
So whenever the trajectory_setpoint is updated, it needs to distinguish if it's in offboard to select the correct topic. The only way to remove vehicle_control_mode and the check for offboard in the landdetector is to remove the subscription to trajectory_setpoint

Reply:
@Jaeyoung-Lim, hey I've had some time to think about what we discussed a while ago on the dev call. I believe you suggested to stop the publishing of trajectory setpoints internally in px4 when trajectory setpoints are detected from an offboard source. The problem with that solution is the flight mode requested by the pilot will be overridden always by offboard. The use case I'm looking for is to be able to send offboard setpoints before switching to offboard without having the setpoints from a PX4 auto mode conflict with the offboard setpoints. I believe this is the only solution for that scenario. Also in the case of a failsafe when setpoints aren't received quickly enough, I would like a PX4 auto mode to take over and completely ignore any future offboard setpoints.

Reply:
@benjinne The mechanism I mentioned is already being used for other setpoints such as attitude or rate setpoints. It is also being used for trajectory setpoints when being sent over mavlink. I dont see a need to habdle this differently just for the fastRTPS interface.

 I think what you mentioned (ignoring offboard setpoints) is already achieved with the mavlink interface. Also I dont think the solution as is is scalable(it doesnt include handling rate, attitude and actuator setpoints) and will be easy to break whenever someone implements a new function and it doesnt consider offboard in their code.